core/src/main/java/org/elasticsearch/index/seqno/GlobalCheckpointTracker.java

```diff
@@ -50,7 +57,7 @@
  */
 public class GlobalCheckpointTracker extends AbstractIndexShardComponent {
 
-    private final String allocationId;
+    final String allocationId;
```

Comment:
some methods in this class also have allocationId as parameter, which is a bit unfortunate. Maybe rename this (I don't have a good suggestion right now)?

Reply:
For now I renamed it to `shardAllocationId`. Let me know if you prefer something else.